File: ABC/ABC301-400/ABC338/C/C.nim
import strutils, sequtils
let N = stdin.readLine.parseInt
let Q = stdin.readLine.split().map(parseInt)
let A = stdin.readLine.split().map(parseInt)
let B = stdin.readLine.split().map(parseInt)
var ans = 0
for i in 0..1000000:
  let C = (0..<N).mapIt(Q[it] - A[it] * i)
  if min(C) >= 0:
    ans = max(ans, i + min((0..<N).mapIt(if B[it] == 0: 1000000 else: C[it] div B[it])))
echo ans

File: ABC/ABC301-400/ABC338/C/C.py
N = int(input())
Q = list(map(int, input().split()))
A = list(map(int, input().split()))
B = list(map(int, input().split()))
ans = 0
for i in range(1000001):
	C = [Q[j] - A[j] * i for j in range(N)]
	if min(C) >= 0:
		ans = max(ans, i + min(C[j] // B[j] for j in range(N) if B[j] != 0))
print(ans)